pkg/limits/frontend/service.go

```diff
@@ -49,15 +52,41 @@ type RingIngestLimitsService struct {
 	pool *ring_client.Pool
 
 	limits Limits
+
+	// metrics
+	tenantExceedsLimits         *prometheus.CounterVec
+	tenantActiveStreams         *prometheus.GaugeVec
+	tenantDuplicateStreamsFound *prometheus.CounterVec
+	tenantRejectedStreams       *prometheus.CounterVec
 }
 
 // NewRingIngestLimitsService returns a new RingIngestLimitsClient.
-func NewRingIngestLimitsService(ring ring.ReadRing, pool *ring_client.Pool, limits Limits, logger log.Logger) *RingIngestLimitsService {
+func NewRingIngestLimitsService(ring ring.ReadRing, pool *ring_client.Pool, limits Limits, logger log.Logger, reg prometheus.Registerer) *RingIngestLimitsService {
 	return &RingIngestLimitsService{
 		logger: logger,
 		ring:   ring,
 		pool:   pool,
 		limits: limits,
+		tenantExceedsLimits: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Namespace: constants.Loki,
+			Name:      "ingest_limits_frontend_exceeds_limits",
+			Help:      "The total number of exceeds limits requests per tenant.",
+		}, []string{"tenant"}),
+		tenantActiveStreams: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
+			Namespace: constants.Loki,
+			Name:      "ingest_limits_frontend_active_streams",
```

Comment:
It might be better to invert the names to make discovering the metrics easier. For example:

```
ingest_limits_frontend_streams_active
ingest_limits_frontend_streams_duplicate_total
ingest_limits_frontend_streams_rejected_total
```

Then it's possible to find all frontend stream metrics searching/autocompleting the prefix `ingest_limits_frontend_streams`.